orchagent/vxlanorch.h

```diff
@@ -68,23 +98,51 @@ struct nh_tunnel_t
     int             ref_count;
 };
 
+typedef enum {
+  TNL_SRC_IMR,
+  TNL_SRC_MAC,
+  TNL_SRC_IP
+} tunnel_user_type_e;
+
+class VxlanTunnel;
+
+typedef struct tunnel_refcnts_s
+{
+   uint32_t      imr_refcnt;
+   uint32_t      mac_refcnt;
+   uint32_t      ip_refcnt;
+   uint32_t      spurious_add_imr_refcnt;
+   uint32_t      spurious_del_imr_refcnt;
+   VxlanTunnel*  dip_tunnel;
+} tunnel_refcnt_t;
+
+typedef struct tunnel_map_entry_s
+{
+   sai_object_id_t map_entry_id;
+   uint32_t        vlan_id;
+   uint32_t        vni_id;
+} tunnel_map_entry_t;
+
 typedef std::map<uint32_t, std::pair<sai_object_id_t, sai_object_id_t>> TunnelMapEntries;
 typedef std::unordered_map<nh_key_t, nh_tunnel_t, nh_key_hash> TunnelNHs;
+typedef std::map<std::string, tunnel_refcnt_t> TunnelUsers;
 
 class VxlanTunnel
 {
 public:
-    VxlanTunnel(string name, IpAddress srcIp, IpAddress dstIp)
-                :tunnel_name_(name), src_ip_(srcIp), dst_ip_(dstIp) { }
+    VxlanTunnel(string name, IpAddress srcIp, IpAddress dstIp, tunnel_creation_src_t src);
```

Comment:
why the values initiations are removed? 
```
VxlanTunnel(std::string name, IpAddress srcIp, IpAddress dstIp, tunnel_creation_src_t src)
           :tunnel_name_(name), src_ip_(srcIp), dst_ip_(dstIp), src_creation_(src) {}
```

Reply:
These have been incorporated in vxlanorch.cpp along with a destructor.